dfs/commit.go

```diff
@@ -48,15 +48,34 @@ func (dfs *DistributedFilesystem) Commit(ctrID string) (string, error) {
 		return "", ErrStaleContainer
 	}
 	// commit the container
+	oldID := rImage.UUID
 	img, err := dfs.docker.CommitContainer(ctr.ID, ctr.Config.Image)
 	if err != nil {
 		glog.Errorf("Could not commit container %s: %s", ctr.ID, err)
 		return "", err
 	}
 	// push the image into the registry
-	if err := dfs.index.PushImage(rImage.String(), img.ID); err != nil {
-		glog.Errorf("Could not push image %s (%s): %s", rImage, img.ID, err)
-		return "", err
+	if (img.ID != oldID) { //If the commit produced a new ID, call PushImageAfterCommit, which will actually push the image twice to address an issue with imageIDs changing in docker 1.9.1
```

Comment:
I don't understand the purpose of this if statement.  The image id ALWAYS changes on a commit.